src/test/kotlin/apply/application/JudgeServiceTest.kt

```diff
@@ -0,0 +1,88 @@
+package apply.application
+
+import apply.COMMIT_HASH
+import apply.application.github.GithubApi
+import apply.createAssignment
+import apply.createCommitResponse
+import apply.createJudgeHistory
+import apply.createMission
+import apply.domain.assignment.AssignmentRepository
+import apply.domain.assignment.getByUserIdAndMissionId
+import apply.domain.judgehistory.JudgeHistoryRepository
+import apply.domain.judgehistory.findLastByUserIdAndMissionId
+import apply.domain.mission.MissionRepository
+import apply.domain.mission.getById
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.BehaviorSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import io.mockk.every
+import io.mockk.mockk
+import java.time.LocalDateTime
+
+class JudgeServiceTest : BehaviorSpec({
+    val missionRepository = mockk<MissionRepository>()
+    val assignmentRepository = mockk<AssignmentRepository>()
+    val judgeHistoryRepository = mockk<JudgeHistoryRepository>()
+    val githubApi = mockk<GithubApi>()
+
+    val judgeService = JudgeService(
+        missionRepository, assignmentRepository, judgeHistoryRepository, githubApi
+    )
+
+    Given("과제 제출 시간이 아닐 때") {
+        every { missionRepository.getById(any()) } returns createMission(
+            startDateTime = LocalDateTime.now().minusMinutes(2),
+            endDateTime = LocalDateTime.now().minusMinutes(1)
+        )
+        every { githubApi.requestCommits(any()) } returns listOf(createCommitResponse())
+        every { assignmentRepository.getByUserIdAndMissionId(any(), any()) } returns createAssignment()
+        every { judgeHistoryRepository.findLastByUserIdAndMissionId(any(), any()) } returns null
+
+        When("예제 테스트를 실행하면") {
+            Then("예제 테스트가 실행되지 않는다.") {
+                shouldThrow<IllegalStateException> { judgeService.runExampleTest(1L, 1L) }
+            }
```

Comment:
c: 다른 서비스 테스트의 공백 컨벤션은 아래와 같이 돼있어서 한번 코멘트 남겨봅니다.
```suggestion
            Then("예제 테스트가 실행되지 않는다.") {
                shouldThrow<IllegalStateException> {
                    judgeService.runExampleTest(1L, 1L)
                }
            }
```